pkg/thanos/statefulset.go

```diff
@@ -530,3 +536,62 @@ func mountSecret(secretSelector *v1.SecretKeySelector, volumeName string, trVolu
 	})
 	return mountpath + "/" + path
 }
+
+func intersection(a, b []string) (i []string) {
+	m := make(map[string]struct{})
+
+	for _, item := range a {
+		m[item] = struct{}{}
+	}
+
+	for _, item := range b {
+		if _, ok := m[item]; ok {
+			i = append(i, item)
+		}
+
+		negatedItem := strings.TrimPrefix(item, "no-")
+		if item == negatedItem {
+			negatedItem = fmt.Sprintf("no-%s", item)
+		}
+
+		if _, ok := m[negatedItem]; ok {
+			i = append(i, item)
+		}
+	}
+	return i
+}
+
+func extractArgKeys(args []monitoringv1.Argument) []string {
+	var k []string
+	for _, arg := range args {
+		key := arg.Name
+		k = append(k, key)
+	}
+
+	return k
+}
+
+func buildArgs(args []monitoringv1.Argument, additionalArgs []monitoringv1.Argument) ([]string, error) {
+	var containerArgs []string
+
+	argKeys := extractArgKeys(args)
+	additionalArgKeys := extractArgKeys(additionalArgs)
+
+	i := intersection(argKeys, additionalArgKeys)
+	if len(i) > 0 {
+		return nil, errors.Errorf("can't set arguments which are already managed by the operator: %s", strings.Join(i, ","))
+	}
+
+	args = append(args, additionalArgs...)
+
+	for _, arg := range args {
+		if arg.Value != "" {
+			containerArgs = append(containerArgs, fmt.Sprintf("--%s=%s", arg.Name, arg.Value))
+		} else {
+			containerArgs = append(containerArgs, fmt.Sprintf("--%s", arg.Name))
+
+		}
+	}
+
+	return containerArgs, nil
+}
```

Comment:
This is a copy of the same functions from `pkg/prometheus/statefulset.go`. I'd rather have them elsewhere than copy them around, suggestions welcome :)

Reply:
I think `pkg/operator` is better place for common functions and methods

Reply:
Thanks @slashpai! Moved it there 👍